swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/AmazonAutoScalingGroupHandler.kt

```diff
@@ -120,12 +119,15 @@ class AmazonAutoScalingGroupHandler(
     }
   }
 
-  override fun handles(workConfiguration: WorkConfiguration): Boolean
-    = workConfiguration.resourceType == SERVER_GROUP && workConfiguration.cloudProvider == AWS && !rules.isEmpty()
+  override fun handles(workConfiguration: WorkConfiguration): Boolean = workConfiguration.resourceType == SERVER_GROUP && workConfiguration.cloudProvider == AWS && !rules.isEmpty()
 
   override fun getCandidates(workConfiguration: WorkConfiguration): List<AmazonAutoScalingGroup>? {
     val params = Parameters(
-      mapOf("account" to workConfiguration.account.accountId!!, "region" to workConfiguration.location)
+      mapOf(
+        "account" to workConfiguration.account.accountId!!,
+        "region" to workConfiguration.location,
+        "environment" to workConfiguration.account.environment!!
```

Comment:
Seems like the core of this is around the parameters objects and maybe generic accounts vs aws edda-enabled accounts. I agree, this can and should be better.

Reply:
I'd like to see the Parameters(Map<String,Any>) class go away in favor of typed constructors. For now, I'm going to remove nullibility from the Account.environment property I added.